xla/service/collective_quantizer.cc

```diff
@@ -50,6 +50,108 @@ auto ConvertToNarrowerType() {
   return m::Convert().WithPredicate(converts_to_narrower_type);
 }
 
+// Matches a type conversion to a type with a larger byte size than that of the
+// operand.
+auto ConvertToWiderType() {
```

Comment:
Nit: IsConvertToWiderType?

Reply:
This is meant to replace m::Convert in Match calls. Let me know if you want to rename it anyway.